pkg/tsdb/elasticsearch/client/ppl_request_test.go

```diff
@@ -0,0 +1,76 @@
+package es
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestPPLRequest(t *testing.T) {
+	Convey("Test elasticsearch PPL request", t, func() {
+		timeField := "@timestamp"
+		index := "default_index"
+		Convey("Given new PPL request builder", func() {
+			b := NewPPLRequestBuilder(index)
+
+			Convey("When building PPL request", func() {
+				pr, err := b.Build()
+				So(err, ShouldBeNil)
+
+				Convey("When marshal to JSON should generate correct json", func() {
+					body, err := json.Marshal(pr)
+					So(err, ShouldBeNil)
+					json, err := simplejson.NewJson(body)
+					So(err, ShouldBeNil)
+					So(json.Get("query").Interface(), ShouldEqual, "")
+				})
+			})
+
+			Convey("When adding default query", func() {
+				b.AddPPLQueryString(timeField, "$timeTo", "$timeFrom", "")
+
+				Convey("When building PPL request", func() {
+					pr, err := b.Build()
+					So(err, ShouldBeNil)
+
+					Convey("Should have query string filter", func() {
+						f := pr.Query
+						So(f, ShouldEqual, "source = default_index | where `@timestamp` > timestamp('$timeFrom') and `@timestamp` < timestamp('$timeTo')")
+					})
+
+					Convey("When marshal to JSON should generate correct json", func() {
+						body, err := json.Marshal(pr)
+						So(err, ShouldBeNil)
+						json, err := simplejson.NewJson(body)
+						So(err, ShouldBeNil)
+						So(json.Get("query").Interface(), ShouldEqual, "source = default_index | where `@timestamp` > timestamp('$timeFrom') and `@timestamp` < timestamp('$timeTo')")
+					})
+				})
+			})
+			Convey("When adding PPL query", func() {
```

Comment:
what is the expected behaviour if the query already have timestamp filter?

Reply:
It would filter out the data such that both timestamp filters are satisfied. 
Are there specific PPL commands users can use to filter out timestamp that might not work on top of this timestamp filtering?